### PR TITLE
itemstats: Add variable healing food

### PR DESCRIPTION
**Pull request by Abextm**
Supersedes #4092

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/itemstats/RangeStatBoost.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+		return r;
+	}
+
+	private String concat(String a, String b)
+	{
+		// If they share a operator, strip b's duplicate
+		if (a.length() > 1 && b.length() > 1)
+		{
+			char a0 = a.charAt(0);
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemstats/RangeStatBoost.java]: final

File: runelite-client/src/main/java/net/runelite/client/plugins/itemstats/RangeStatBoost.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		final StatChange a = this.a.effect(client);
+		final StatChange b = this.b.effect(client);
+
+		StatChange r = new StatChange();
+		r.setAbsolute(concat(a.getAbsolute(), b.getAbsolute()));
+		r.setRelative(concat(a.getRelative(), b.getRelative()));
+		r.setTheoretical(concat(a.getTheoretical(), b.getTheoretical()));
+		r.setStat(a.getStat());
+
+		int avg = (a.getPositivity().ordinal() + b.getPositivity().ordinal()) / 2;
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemstats/RangeStatBoost.java]: final

File: runelite-client/src/main/java/net/runelite/client/plugins/itemstats/RangeStatBoost.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+		this.b = b;
+	}
+
+	@Override
+	public StatChange effect(Client client)
+	{
+		final StatChange a = this.a.effect(client);
+		final StatChange b = this.b.effect(client);
+
+		StatChange r = new StatChange();
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemstats/RangeStatBoost.java]: final

**Comment by permedd** [comment]
I dunno how this works without pictures but this is awesome also when you hoover your mouse on top of the food it will show how much hp it will give but can it also show what your hp will be if you do eat the food?

**Comment by deathbeam** [comment]
screenshot 2018-07-04 23 32 24

and for what HP you will be on i guess you can configure that in item stats already.

**Pull request merged**